handler/deployment/deployment.go

```diff
@@ -198,95 +188,68 @@ func (d *Deployment) run(ctx context.Context, msg *engine.Message) error {
 	if err := d.post(fmt.Sprintf(runURL, d.endpoint), msg, &res); err != nil {
 		return err
 	}
-	if err := d.monitor(ctx, msg.ID, res.EvalID); err != nil {
+	if err := d.monitor(ctx, msg.ID, res.EvalID, msg.Service, res.JobModifyIndex); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (d *Deployment) monitor(ctx context.Context, deploymentID, evaluationID string) error {
-	if err := d.evaluation(ctx, deploymentID, evaluationID); err != nil {
-		return err
-	}
-	if err := d.allocations(ctx, deploymentID, evaluationID); err != nil {
+func (d *Deployment) monitor(ctx context.Context, correlationID, evaluationID, jobID string, jobSpecModifyIndex uint64) error {
+	if err := d.deploymentSuccess(ctx, correlationID, evaluationID, jobID, jobSpecModifyIndex); err != nil {
```

Comment:
Now that we only need to monitor the deployment this functions feels unnecessary.  Can we just collapse this and the function below?